### DISCUSSION TITLE
Using register( "firstName" , { values:  "abc" } )  seems to fix the watch() to "abc"

### DISCUSSION THREAD

**Discussion by andrwo** [react-hook-form/react-hook-form#6137] [Q&A] [answered]
Seeking some help. I need to use `register()` to register my form field, and also set a default value. 

```jsx
  const fn = register("firstName", { value: "Andrew" });
  const vGetValues = getValues("firstName");
  const vWatch = watch("firstName");
  console.log("getValues:", vGetValues, "... watch:", vWatch);
```
However, I found that registering the default value this way, both `watch()` and `getValues()` does not seem to return the right value (at the render cycle, so using getValues() may not be too appropriate) -- the display at the `console.log()` seems stuck at "Andrew" even if I change it at the UI itself. The form submit however works properly though.

The CodeSandbox that illustrates this issue (just change the FIrstName and observe the console.log):

https://codesandbox.io/s/react-hook-form-js-forked-psqm0?file=/src/App.js

This behaviour is not exhibited in other ways of setting the `defaultValue` (e.g. pass into `Controller` or in `useForm()`). However I need to use this `register( xx, { value: v } )` format in my code for some custom fields.

Thanks in advance!

**Comment by bluebill1049** [comment, marked as answer]
`watch` has an inline condition to return `defaultValue` before mounted, you will have to supply inline defaultValue instead.

**Reply by andrwo** [reply]
The issue now is after mounted. The `watch` value is still stuck at the the `register()` value (that is: stuck at "Andrew"), despite me typing changes on the field in the UI.

**Reply by bluebill1049** [reply]
that's because each time that register gets invoked the value is get updated again. `value` was made for something static.

**Reply by bluebill1049** [reply]
it's not **defaultValue**.

**Reply by andrwo** [reply]
I see. Is is there any possibility to have a `defaultValue` to cater for this scenario?

I would think that this would be the more useful mode as it also caters also to the fixed value scenario if it never changes. Especially for functional components.

**Reply by bluebill1049** [reply]
I think you can put register into useEffect tho

**Reply by andrwo** [reply]
Ic, ok thanks!
